{"uid":"dc480f766ad8b22","name":"Проверка отсутствия прав на создание проекта","fullName":"tests.backend.test_create_project.TestProjectCreate#test_negative_creating_project_by_roles","historyId":"1d676ecbf52365fda97e5a00b4ed6320","time":{"start":1711454307393,"stop":1711454307393,"duration":0},"description":"Проверяется невозможность создания проекта под определенными ролями","descriptionHtml":"<p>Проверяется невозможность создания проекта под определенными ролями</p>\n","status":"broken","statusMessage":"ValueError: Unexpected status code: 401, expected: 200","statusTrace":"self = <api.auth_api.AuthAPI object at 0x7f8d328b7fa0>\nuser_creds = ('', '8976389362930650481')\n\n    def auth_and_get_csrf(self, user_creds):\n        self.session.auth = user_creds\n        try:\n>           csrf_token = self.send_request(\"GET\", \"/authenticationTest.html?csrf\").text\n\napi/auth_api.py:9: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.auth_api.AuthAPI object at 0x7f8d328b7fa0>, method = 'GET'\nendpoint = '/authenticationTest.html?csrf', data = None\nexpected_status = <HTTPStatus.OK: 200>, need_logging = True\nneed_verify_status = True\n\n    def send_request(self, method, endpoint, data=None, expected_status=HTTPStatus.OK, need_logging=True,\n                     need_verify_status=True):\n        \"\"\"\n        Врапер для запросов. Позволяет прикручивать дополнительную логику\n        :param method: Метод запроса\n        :param endpoint: Ендпоинт для склейки с BASE_URL\n        :param data: Тело запроса. По умолчанию пустое, чтобы пропускало no-content запросы\n        :param expected_status: Ожидаемый статус код\n        :param need_logging: Флаг для логирования. По умолчанию = True\n        :param need_verify_status: Флаг для проверки статус кода в ответе. По умолчанию = True\n        :return: Возвращает объект ответа\n        \"\"\"\n        url = f\"{self.base_url}{endpoint}\"\n        response = self.session.request(method=method, url=url, json=data)\n        if need_logging:\n            self.log_request_and_response(response, expected_status, need_verify_status)\n        if need_verify_status:\n            if response.status_code != expected_status:\n>               raise ValueError(f\"Unexpected status code: {response.status_code}, expected: {expected_status}\")\nE               ValueError: Unexpected status code: 401, expected: 200\n\ncustom_requester/custom_requester.py:34: ValueError\n\nDuring handling of the above exception, another exception occurred:\n\nuser_session = <function user_session.<locals>._create_user_session at 0x7f8d327416c0>\n\n    @pytest.fixture\n    def super_admin(user_session):\n        new_session = user_session()\n        super_admin = User(SuperAdminCreds.USERNAME, SuperAdminCreds.PASSWORD, None, new_session,\n                           [\"SUPER_ADMIN\"])\n>       super_admin.api_manager.auth_api.auth_and_get_csrf(super_admin.creds)\n\nconftest.py:51: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.auth_api.AuthAPI object at 0x7f8d328b7fa0>\nuser_creds = ('', '8976389362930650481')\n\n    def auth_and_get_csrf(self, user_creds):\n        self.session.auth = user_creds\n        try:\n            csrf_token = self.send_request(\"GET\", \"/authenticationTest.html?csrf\").text\n        except Exception as e:\n>           raise ValueError(f\"{e}\")\nE           ValueError: Unexpected status code: 401, expected: 200\n\napi/auth_api.py:11: ValueError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_session_faker","time":{"start":1711454307088,"stop":1711454307111,"duration":23},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"super_admin","time":{"start":1711454307394,"stop":1711454307402,"duration":8},"status":"broken","statusMessage":"ValueError: Unexpected status code: 401, expected: 200\n","statusTrace":"  File \"/home/runner/.local/lib/python3.10/site-packages/pluggy/_callers.py\", line 166, in _multicall\n    teardown.throw(outcome._exception)\n  File \"/home/runner/.local/lib/python3.10/site-packages/_pytest/setuponly.py\", line 36, in pytest_fixture_setup\n    return (yield)\n  File \"/home/runner/.local/lib/python3.10/site-packages/pluggy/_callers.py\", line 102, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/.local/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1149, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/home/runner/.local/lib/python3.10/site-packages/_pytest/fixtures.py\", line 923, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n  File \"/home/runner/work/TeamCity_Test/TeamCity_Test/conftest.py\", line 51, in super_admin\n    super_admin.api_manager.auth_api.auth_and_get_csrf(super_admin.creds)\n  File \"/home/runner/work/TeamCity_Test/TeamCity_Test/api/auth_api.py\", line 11, in auth_and_get_csrf\n    raise ValueError(f\"{e}\")\n","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false},{"name":"user_session","time":{"start":1711454307394,"stop":1711454307394,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Проверяется невозможность создания проекта под определенными ролями","status":"broken","statusMessage":"ValueError: Unexpected status code: 401, expected: 200","statusTrace":"self = <api.auth_api.AuthAPI object at 0x7f8d328b7fa0>\nuser_creds = ('', '8976389362930650481')\n\n    def auth_and_get_csrf(self, user_creds):\n        self.session.auth = user_creds\n        try:\n>           csrf_token = self.send_request(\"GET\", \"/authenticationTest.html?csrf\").text\n\napi/auth_api.py:9: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.auth_api.AuthAPI object at 0x7f8d328b7fa0>, method = 'GET'\nendpoint = '/authenticationTest.html?csrf', data = None\nexpected_status = <HTTPStatus.OK: 200>, need_logging = True\nneed_verify_status = True\n\n    def send_request(self, method, endpoint, data=None, expected_status=HTTPStatus.OK, need_logging=True,\n                     need_verify_status=True):\n        \"\"\"\n        Врапер для запросов. Позволяет прикручивать дополнительную логику\n        :param method: Метод запроса\n        :param endpoint: Ендпоинт для склейки с BASE_URL\n        :param data: Тело запроса. По умолчанию пустое, чтобы пропускало no-content запросы\n        :param expected_status: Ожидаемый статус код\n        :param need_logging: Флаг для логирования. По умолчанию = True\n        :param need_verify_status: Флаг для проверки статус кода в ответе. По умолчанию = True\n        :return: Возвращает объект ответа\n        \"\"\"\n        url = f\"{self.base_url}{endpoint}\"\n        response = self.session.request(method=method, url=url, json=data)\n        if need_logging:\n            self.log_request_and_response(response, expected_status, need_verify_status)\n        if need_verify_status:\n            if response.status_code != expected_status:\n>               raise ValueError(f\"Unexpected status code: {response.status_code}, expected: {expected_status}\")\nE               ValueError: Unexpected status code: 401, expected: 200\n\ncustom_requester/custom_requester.py:34: ValueError\n\nDuring handling of the above exception, another exception occurred:\n\nuser_session = <function user_session.<locals>._create_user_session at 0x7f8d327416c0>\n\n    @pytest.fixture\n    def super_admin(user_session):\n        new_session = user_session()\n        super_admin = User(SuperAdminCreds.USERNAME, SuperAdminCreds.PASSWORD, None, new_session,\n                           [\"SUPER_ADMIN\"])\n>       super_admin.api_manager.auth_api.auth_and_get_csrf(super_admin.creds)\n\nconftest.py:51: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <api.auth_api.AuthAPI object at 0x7f8d328b7fa0>\nuser_creds = ('', '8976389362930650481')\n\n    def auth_and_get_csrf(self, user_creds):\n        self.session.auth = user_creds\n        try:\n            csrf_token = self.send_request(\"GET\", \"/authenticationTest.html?csrf\").text\n        except Exception as e:\n>           raise ValueError(f\"{e}\")\nE           ValueError: Unexpected status code: 401, expected: 200\n\napi/auth_api.py:11: ValueError","steps":[],"attachments":[{"uid":"f9a08332e45728bf","name":"log","source":"f9a08332e45728bf.txt","type":"text/plain","size":628}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"user_session::0","time":{"start":1711454307420,"stop":1711454307420,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"subSuite","value":"Негативные сценарии"},{"name":"suite","value":"Управление проектами"},{"name":"severity","value":"normal"},{"name":"parentSuite","value":"API тесты"},{"name":"tag","value":"projects"},{"name":"host","value":"fv-az529-345"},{"name":"thread","value":"7763-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.backend.test_create_project"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"role","value":"'PROJECT_VIEWER'"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":2,"skipped":0,"passed":3,"unknown":0,"total":5},"items":[{"uid":"9743a560fefea06e","reportUrl":"https://QAKonstantin.github.io/TeamCity_Test/122//#testresult/9743a560fefea06e","status":"broken","statusDetails":"ValueError: Unexpected status code: 401, expected: 200","time":{"start":1711452159940,"stop":1711452159940,"duration":0}},{"uid":"17730d7c47df1631","reportUrl":"https://QAKonstantin.github.io/TeamCity_Test/121//#testresult/17730d7c47df1631","status":"passed","time":{"start":1711451063287,"stop":1711451063422,"duration":135}},{"uid":"27d2b1a3870c6484","reportUrl":"https://QAKonstantin.github.io/TeamCity_Test/120//#testresult/27d2b1a3870c6484","status":"passed","time":{"start":1711369184101,"stop":1711369184253,"duration":152}},{"uid":"f4babbd88498e73","reportUrl":"https://QAKonstantin.github.io/TeamCity_Test/119//#testresult/f4babbd88498e73","status":"passed","time":{"start":1711356085051,"stop":1711356085179,"duration":128}}]},"tags":["projects"]},"source":"dc480f766ad8b22.json","parameterValues":["'PROJECT_VIEWER'"]}